src/components/StatsPanel/StatsPanel.hook.ts

```diff
@@ -0,0 +1,20 @@
+import { useMemo } from 'react';
+
+import { TEST_STATS } from './StatsPanel.constants';
+
+import type { StatItemData, StatsPanelProps } from './StatsPanel.types';
+export const useStatsPanel = ({
```

Comment:
The hook destructures StatsPanelProps directly (no defaults). While this is OK if callers always provide the props, it could be made more defensive. Also the hook relies on TEST_STATS keys always existing; a mismatch between TEST_STATS keys and the destructured names could cause runtime issues.

**Sugerencia:**
```typescript
Consider providing default values or validating inputs in the hook and give TypeScript stronger guarantees on TEST_STATS (see suggestion in constants). Example:

export const useStatsPanel = ({ totalTests = 0, passed = 0, failed = 0, waiting = 0 }: StatsPanelProps): StatItemData[] =>
  useMemo(() => [...], [totalTests, passed, failed, waiting]);

Also ensure TEST_STATS is typed to guarantee 'total', 'passed', 'failed', 'waiting' keys exist.
```